functional_programming_tests/test/with_dependency_injection.cpp

```diff
@@ -0,0 +1,528 @@
+            std::shared_ptr<example_srvs::srv::SetMap::Response> response) {
+          // Set the map to generate the path from
+          response->success.data = costmap_setter(request->map);
+        });
+    // Register the generate path service
+    mw_->register_generate_path_service(  // Trivial comment to enfore params on
+                                          // new line
+        [this](
+            std::shared_ptr<example_srvs::srv::GetPath::Request> const request,
+            std::shared_ptr<example_srvs::srv::GetPath::Response> response) {
+          if (map_.get_data().size() == 0) {
+            mw_->log_error("MAP IS EMPTY!!");
+            response->success.data = false;
+            response->path = std_msgs::msg::UInt8MultiArray();
+            return;
+          }
+          // Check to make sure start and goal fields of the request are of
+          // size 2
+          if (request->start.data.size() != 2) {
+            mw_->log_error("START POSITION MUST CONTAIN TWO ELEMENTS!!");
+            response->success.data = false;
+            response->path = std_msgs::msg::UInt8MultiArray();
+            return;
+          }
+          if (request->goal.data.size() != 2) {
+            mw_->log_error("GOAL POSITION MUST CONTAIN TWO ELEMENTS!!");
+            response->success.data = false;
+            response->path = std_msgs::msg::UInt8MultiArray();
+            return;
+          }
+          auto const start =
+              Position{request->start.data[0], request->start.data[1]};
+          auto const goal =
+              Position{request->goal.data[0], request->goal.data[1]};
+
+          // Generate the path
+          auto const path = generate_global_path(start, goal);
+
+          // Start populating the response message
+          auto response_path = std_msgs::msg::UInt8MultiArray();
+
+          if (path.has_value()) {
+            response_path.layout.dim.resize(
+                3, std_msgs::msg::MultiArrayDimension());
+
+            response_path.layout.dim[0].label = "rows";
+            response_path.layout.dim[0].size = path.value().size();
+            response_path.layout.dim[0].stride = path.value().size() * 2;
+
+            response_path.layout.dim[1].label = "columns";
+            response_path.layout.dim[1].size = 2;
+            response_path.layout.dim[1].stride = 1;
+
+            response_path.layout.dim[2].label = "channel";
+            response_path.layout.dim[2].size = 1;
+            response_path.layout.dim[2].stride = 1;
+
+            // Start pushing back the path only if there is one
+            if (path.value().size() > 0) {
+              for (auto const& position : path.value()) {
+                response_path.data.push_back(position.x);
+                response_path.data.push_back(position.y);
+              }
+            }
+          }
+
+          response->success.data = path.has_value();
+          response->path = response_path;
+        });
```

Comment:
As I think we've discussed now, moving it in to it's own function would be just fine. I didn't do it here because it just wasn't what I was highlighting.